client/lib/titan/get-max-titan-mailbox-count.ts

```diff
@@ -0,0 +1,5 @@
+import type { ResponseDomain } from 'calypso/lib/domains/types';
+
+export function getMaxTitanMailboxCount( domain: ResponseDomain | undefined ): number {
+	return domain?.titanMailSubscription?.maximumMailboxCount ?? 1;
```

Comment:
I'm realizing that `?? 1` differs from what we had on L183 in `client/my-sites/email/email-management/email-home.tsx` previously. It seems strange to me that we would default to returning `1` from this function. It also doesn't match the behavior of `getGSuiteMailboxCount`. Perhaps we could change this function to default to return `0`?

Reply:
Yes, that's surprising. Maybe @olaseni remembers the reason for that.

Reply:
The maximum number of mailboxes per domain is often set at provisioning time, and is represented by `titanMailSubscription.maximumMailboxCount` in `getMaxTitanMailboxCount()`. This number cannot be less than 1. This is not the actual number of mailboxes a domain/user has, but the maximum number they **_can_** have. It serves as a limiting number, and a user can not purchase more than this number.

`getConfiguredTitanMailboxCount()` is the figure that corresponds to the number of existing mailboxes at a time. That  defaults to zero as you can see.

Perhaps the confusion comes from the way they are named?

Reply:
Sure, but if `domain?.titanMailSubscription?.maximumMailboxCount` is null-ish, the reason for that might be that the domain doesn't have an active Titan subscription. Wouldn't it make sense to return `0` as the default then, rather than `1`?

Reply:
I think it depends on how you look at it. That number is a configuration figure that defaults to 1 even before a purchase  happens and doesn't depend on a subscription.

BTW, all references in the codebase guard that there's a subscription before accessing that value:

```js
if ( hasTitanMailWithUs( domain ) ) {
	const titanMailboxCount = getMaxTitanMailboxCount( domain );
}	
```

Reply:
We discussed this separately in Slack and concluded that the current default return value of `getMaxTitanMailboxCount` is primarily adapted for scenarios where the user intends to purchase a Titan subscription for the domain. Because Titan subscriptions can't have fewer than one mailboxes, `1` is the default return value. 

This might change in the future, but for now, we'll leave it as is. I included a comment in the code to clarify this though.